MIT-LCP/physionet-build
# create training basics  --> implement take training feature and resume training feature for user

## What?
Here i have added a feature that will allow users to take the training from settings. User can choose from a list of available 
trainings and take the onplatform training. Users can also see the progress of the training and can resume the training from where they left off.

## Why?
This is a follow up PR for https://github.com/MIT-LCP/physionet-build/pull/1950

## How?
Here we create 3 new views to handle the training.

**1. `training.views.take_training`**

This view will take care of training homepage. It will show progress of the training if the user has already started the training, show them details of the training and allow them to start/resume and review the training.


**2.  `training.views.take_module_training`**

This view will take care of letting users take individual modules of the training. It will allow users to take the module from where they left off.
Users will need to finish module1 before they can start module2 and so on.

Here we have used a considerable amount of javascript to make the training interactive. How it works is when the user loads the module page,
we initially load all the content, quizzes and order them based on the `order` field, and initially we hide all the content and quizzes.

Then we show either the first content,quiz or the next one after a content, quiz that they had completed last time. and we keep doing this until the user has completed the module.

**3. `training.views.update_module_progress`**

    This view uses a simple post request and expects an ajax request from the client. It will update the progress of the module and the progress of the training.


## Testing? [WIP]

Following tests will be added
1. Check access to the training homepage
2. Check access to the training module page
3. Check valid and invalid submission of the training module
4. Check valid and invalid submission of the training


## Screenshots (optional)

![image](https://user-images.githubusercontent.com/24412619/228581570-568f28bc-28d3-4e22-a699-bd05092f77f2.png)

![image](https://user-images.githubusercontent.com/24412619/228581290-4615bc2e-8fbe-4aaf-9706-e19f34e7e1e0.png)

![image](https://user-images.githubusercontent.com/24412619/228581693-d42ab459-9563-49ee-98a0-bb5b61dfce2f.png)

![image](https://user-images.githubusercontent.com/24412619/228581764-e4368b6d-755c-460e-9496-0252c109a449.png)


## Anything Else?

### How to take a training as a user?

1. Login as user
2. Navigate to `Settings`
3. Select the training from the dropdown
4. Start the training
